lib/ESP/prusa/src/system/esp_tcp.c

```diff
@@ -669,21 +697,15 @@ static err_t altcp_esp_close(struct altcp_pcb *conn) {
     esp_pcb *epcb = (esp_pcb *)conn->state;
     if (!epcb) {
         ALTCP_ESP_DEBUG_FN("epcb is NULL !!!");
-        return espERR;
+        return ERR_VAL;
     }
     ASSERT_EPCB(epcb);
 
-    ALTCP_ESP_DEBUG_FN("Closing connection: %d, total %ld packets, %ld bytes", esp_conn_getnum(epcb->econn), epcb->rcv_packets, epcb->rcv_bytes);
-    espr_t err = esp_conn_close(epcb->econn, 0);
-
-    if (err != espOK) {
-        ALTCP_ESP_DEBUG_FN("Failed to close connection: %d", err);
-        return espr_t2err_t(err);
+    if (epcb->listen_port) {
+        return close_listen_mode(epcb);
     }
 
-    epcb->closing = 1;
-
-    return ERR_OK;
+    return close_normal_mode(epcb);
```

Comment:
Minor style nitpick: I'd prefer ` else { ` here instead of relying on that return above exiting the function.

Reply:
Fixed, I admit it looks much better with explicit else. (One more point for disabling no-else-return check in pylint)